workflow/controller/operator.go

```diff
@@ -388,7 +388,7 @@ func (woc *wfOperationCtx) operate() {
 		}
 	}
 
-	err = woc.deletePVCs()
+	err = woc.deletePVCs(workflowStatus)
```

Comment:
huh? how come woc.wf.Status.Phase is not suitable?

Reply:
At this point `woc.wf.Status.Phase` hasn't been changed. This happens in the `switch` block on line 414 onwards.

Reply:
I think this block of code should be moved from line 391 to line 443

Reply:
I think we would therefore need to change how to deal with the error scenario - we should not mark a workflow as success then changes to error. Probably just log it.

Reply:
[fix-argo-retry-with-pvcs 6c93455d] Move woc.deletePVCs() to after workflow phase has been updated